timApp/plugin/jsrunner/util.py

```diff
@@ -690,3 +708,40 @@ def _handle_item_right_actions(
 
     # Flush so that access rights are correct for any other JSRunner operations
     db.session.flush()
+
+
+def _handle_mail_to_send(mail: list[MailToSendData]) -> None:
+    mail_to_send: dict[tuple[str, str], dict[str, Any]] = {}
+    user_cache: dict[str, User | None] = {}
+
+    # Try to merge mails with the same subject and body
```

Comment:
Piti kommentoida pr mutta tää meni vahingossa suoraan tonne commitin kommentteihin:


Nyt ei ihan auennu mitä täs pitäis tapahtuu. Jos laitan yhdelle henkilölle esim 3 postia, joissa on kaikissa sama otsikko mutta eri sisältö, niin niiden kaikkien sisältö muuttuu vastaamaan sen ekan postin sisältöä. Silti se viesti lähetetään kolmesti

![kuva](https://github.com/TIM-JYU/TIM/assets/63715895/a4522b87-cd61-4402-917a-08f499f46f71)

![kuva](https://github.com/TIM-JYU/TIM/assets/63715895/162c5ba5-5de6-438a-abab-3b342ead5be7)


Tai jos laittaa 2 postia joissa molemmissa on sama kohde, otsikko ja sisältö niin silti sekin tulee kahdesti

Reply:
Ääh, tuohon jäi paha typo. Avaimena pitää olla siis `(subject, body)`, jotta samaa otsikkoa ja viestiä sisältävät postit yhdistettäisiin yhteen.
Omassa testauksessakaan en tuota huomannut, sillä nykyinen `multi_send_email` lähettää jokaiselle vastaanottajalle oman erillisen postin yhden massapostin sijaan. 

Nyt on korjattu ja selkeytetty ajatus. Eli siis jos samalle tai useammalle vastaanottajalle lähetetään sama posti (sama otsikko ja sisältä) yksi tai monta kertaa, ne mergetään yhdeksi massapostiksi.

timdevsissa on nyt korjattu versio pyörimässä

Reply:
Jes, nyt tuli eri sisältöiset oikein ja duplikaatit tuli yhtenä